### PR TITLE
[Example 15] Using multiple controller managers under different namespaces

### PR DESCRIPTION
**Pull request by christophfroehlich**
A rewrite of #170 including https://github.com/StoglRobotics-forks/ros2_control_demos/pull/8

An open questions for me:
- It uses the rrbot_base.launch.py: Should we move it to the description package? I deleted it once for the other examples for sake of simplicity. But here it makes sense IMHO.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,10 @@ The following examples are part of this demo repository:
 
 * Example 14: ["Modular robots with actuators not providing states and with additional sensors"](example_14)
 
+* Example 15: ["Using multiple controller managers under different namespaces"](example_15)
+
+   This example shows how to include multiple robots in namespaced controller manager instances.
```
Comment (saikishor) [inline README.md:75]:
```suggestion
* Example 14: ["Modular robots with actuators not providing states and with additional sensors"](example_14)

   The example shows how to implement robot hardware with actuators not providing states and with additional sensors.

* Example 15: ["Using multiple controller managers"](example_15)

   This example shows how to integrate multiple robots under different controller manager instances.
```

File: doc/index.rst [hunk 2]
```diff
@@ -78,6 +78,10 @@ Example 12: "Controller chaining"
 Example 13: "Multi-robot example (tba.)"
 
 Example 14: "Modular robots with actuators not providing states and with additional sensors"
+   The example shows how to implement robot hardware with actuators not providing states and with additional sensors.
+
+Example 15: "Using multiple controller managers under different namespaces"
+   This example shows how to include multiple robots in namespaced controller manager instances.
```
Comment (saikishor) [inline doc/index.rst:84]:
```suggestion
Example 15: "Using multiple controller managers"
   This example shows how to integrate multiple robots under different controller manager instances.
```

File: doc/index.rst [hunk 3]
```diff
@@ -272,3 +276,4 @@ Examples
    Example 10: Industrial robot with GPIO interfaces <../example_10/doc/userdoc.rst>
    Example 12: Controller chaining <../example_12/doc/userdoc.rst>
    Example 14: Modular robots with actuators not providing states <../example_14/doc/userdoc.rst>
+   Example 15: Using multiple controller managers under different namespaces <../example_15/doc/userdoc.rst>
```
Comment (saikishor) [inline doc/index.rst:279]:
```suggestion
   Example 15: Using multiple controller managers <../example_15/doc/userdoc.rst>
```

File: example_15/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# ros2_control_demo_example_15
+
+   This example shows how to include multiple robots in namespaced controller manager instances.
```
Comment (saikishor) [inline example_15/README.md:3]:
```suggestion
   This example shows how to integrate multiple robots under different controller manager instances.
```

File: example_15/doc/userdoc.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
+:github_url: https://github.com/ros-controls/ros2_control_demos/blob/{REPOS_FILE_BRANCH}/example_15/doc/userdoc.rst
+
+.. _ros2_control_demos_example_15_userdoc:
+
+Example 15: Using multiple controller managers under different namespaces
+=========================================================================
+
+This example shows how to include multiple robots in namespaced controller manager instances.
```
Comment (saikishor) [inline example_15/doc/userdoc.rst:8]:
```suggestion
This example shows how to integrate multiple robots under different controller manager instances.
```

File: example_15/doc/userdoc.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
+:github_url: https://github.com/ros-controls/ros2_control_demos/blob/{REPOS_FILE_BRANCH}/example_15/doc/userdoc.rst
+
+.. _ros2_control_demos_example_15_userdoc:
+
+Example 15: Using multiple controller managers under different namespaces
+=========================================================================
```
Comment (saikishor) [inline example_15/doc/userdoc.rst:6]:
```suggestion
Example 15: Using multiple controller managers
==============================================
```

File: example_15/doc/userdoc.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+This example shows how to include multiple robots in namespaced controller manager instances.
+
+.. include:: ../../doc/run_from_docker.rst
+
+Scenario showcase: Using ros2_control within a local namespace
+----------------------------------------------------------------
```
Comment (saikishor) [inline example_15/doc/userdoc.rst:13]:
```suggestion
Scenario: Using ros2_control within a local namespace
-----------------------------------------------------
```

File: example_15/doc/userdoc.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+  ros2 control switch_controllers -c /rrbot/controller_manager --deactivate forward_position_controller --activate position_trajectory_controller
+
+Commanding the robot using ``JointTrajectoryController`` (name: ``/rrbot/position_trajectory_controller``)
+
+.. code-block:: shell
+
+  ros2 launch ros2_control_demo_example_15 test_joint_trajectory_controller.launch.py publisher_config:=rrbot_namespace_joint_trajectory_publisher.yaml
+
+Scenario showcase: Using multiple controller managers on the same machine
+-------------------------------------------------------------------------
```
Comment (saikishor) [inline example_15/doc/userdoc.rst:79]:
```suggestion
Scenario: Using multiple controller managers on the same machine
----------------------------------------------------------------
```

**Review by saikishor — CHANGES_REQUESTED** [review]
LGTM. The tutorial works too.

Thank you

**Comment by christophfroehlich** [comment]
> LGTM. The tutorial works too. Just some minor changes.
> 
> Thank you

thx. what do you think about the two questions in the PR description?

**Review by olivier-stasse — APPROVED** [review]
Thanks a lot for this great example.
Tested on my machine (22.04 LTS + head of ros2_control and ros2_controllers)

**Comment by olivier-stasse** [comment]
I don't understand which change is missing from @saikishor review.

**Comment by christophfroehlich** [comment]
> I don't understand which change is missing from @saikishor review.

Thanks for your review. I addressed his comments, only the final approval was missing.

**Pull request merged**